src/lib/index.ts

```diff
@@ -19,62 +19,71 @@ export default class BundleChecker {
 
   constructor(params: IBundleCheckerParams) {
     this.inputParams = params; // TODO: perform default override of some params
-    this.workDir = this.generateWorkDirName();
   }
 
+  // Refactor this, it  is doing too much
   public async compare(): Promise<IBundleCheckerReport> {
-    await this.init();
-    const { githubRepo, currentBranch, targetBranch } = this.inputParams;
+    let result: IBundleCheckerReport;
+
+    const { currentBranch, targetBranch } = this.inputParams;
     try {
-      process.chdir(this.workDir);
-      const { stdout } = await exec(`pwd`);
-      console.log(`PWD: ${stdout}`);
-      await this.cloneRepo(githubRepo);
-      // --- CURRENT
+      await this.init();
+      // --- CURRENT BRANCH
       this.spinner.indent = 4;
       this.spinner.info(`Revision: ${currentBranch}`);
       await this.buildBranch(currentBranch);
       const currentSize = await this.getTotalSize();
+
       // --- CLEAN
       this.spinner.indent = 0;
       await this.cleanDist();
-      // --- TARGET
+
+      // --- TARGET BRANCH
       this.spinner.indent = 4;
       this.spinner.info(`Revision: ${targetBranch}`);
       await this.buildBranch(targetBranch);
       const targetSize = await this.getTotalSize();
-      return this.generateReport(`
-      Current: ${JSON.stringify(currentSize)},
-      Target:${JSON.stringify(targetSize)}
-      `);
+      result = this.generateReport(`
+        Current: ${JSON.stringify(currentSize)},
+        Target:${JSON.stringify(targetSize)}`);
     } catch (e) {
       console.log(e);
       console.log(JSON.stringify(e));
       this.spinner.fail(e);
-      return { reportText: '0' };
+      result = { reportText: '0' };
     }
+    await this.destory();
+    return result;
   }
 
   private async init() {
-    // TODO: check if we have all the permissions
-    await this.makeFolder(this.workDir);
+    if (this.inputParams.githubRepo) {
+      this.workDir = this.generateWorkDirName();
+      await exec(`mkdir -p ${this.workDir}`);
+      process.chdir(this.workDir);
+      const { stdout } = await exec(`pwd`);
+      console.log(`PWD: ${stdout}`);
+      await this.cloneRepo(this.inputParams.githubRepo);
+    }
   }
 
-  // private printResult(current: string, target: string) {
-  //   this.spinner.info(`Current: ${printBytes(current)}`);
-  //   this.spinner.info(`Target: ${printBytes(target)}`);
-  //   this.spinner.info(`Diff: ${printBytes(parseInt(current, 10) - parseInt(target, 10))}`);
-  // }
+  private async destory() {
+    if (this.inputParams.githubRepo) {
```

Comment:
minor: could we rename this to `gitRepository` ?

Reply:
yes